pkg/perm/perm.go

```diff
@@ -0,0 +1,61 @@
+package perm
+
+func str(a []string, f func([]string), i int) {
+	if i > len(a) {
+		f(a)
+		return
+	}
+
+	str(a, f, i+1)
+
+	for j := i + 1; j < len(a); j++ {
+		a[i], a[j] = a[j], a[i]
+		str(a, f, i+1)
+		a[i], a[j] = a[j], a[i]
+	}
+}
+
+// Permutations call f on all possible combination of the set.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [comment on exported function Strings should be of the form "Strings ..."](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)